gapic-common/test/gapic/rest/client_stub_test.rb

```diff
@@ -0,0 +1,114 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "test_helper"
+require "gapic/rest"
+
+class ClientStubTest < Minitest::Test
+  ENDPOINT = "google.example.com"
+  CREDENTIALS = :dummy_credentials
+  STANDARD_HEADER = { "Content-Type" => "application/json" }
+
+  FakeOptions = ::Struct.new :timeout
+
+  FakeRequest = ::Struct.new :params, :body, :headers, :options do
+    def initialize
+      self.options = FakeOptions.new
+      self.headers = {}
+    end
+
+    def self.default
+      req = FakeRequest.new
+      req.headers = STANDARD_HEADER.dup
+      req
+    end
+  end
+
+  def make_client_stub numeric_enums: false
+    ::Gapic::Rest::ClientStub.new endpoint: ENDPOINT,
+                                  credentials: CREDENTIALS,
+                                  numeric_enums: numeric_enums
+  end
+
+  def expect_connection client_stub, verb, uri, req
+    mock = ::Minitest::Mock.new
+    client_stub.instance_variable_set :@connection, mock
+    mock.expect verb, nil do |actual_uri, &block|
+      actual_req = FakeRequest.default
+      block.call actual_req
+      uri == actual_uri && req == actual_req
+    end
+  end
+
+  def test_make_get_request_simple
+    client_stub = make_client_stub
+    expected_req = FakeRequest.default
+    mock = expect_connection client_stub, :get, "/blah", expected_req
+    client_stub.make_get_request uri: "/blah"
+    mock.verify
+  end
+
+  def test_make_get_request_with_params
+    client_stub = make_client_stub
+    expected_req = FakeRequest.default
+    expected_req.params = {"Foo" => "bar"}
+    mock = expect_connection client_stub, :get, "/blah", expected_req
+    client_stub.make_get_request uri: "/blah", params: {"Foo" => "bar"}
+    mock.verify
+  end
+
+  def test_make_get_request_with_numeric_enums
```

Comment:
you're not testing the case where the `params` end up already having the `alt` key (first line of your changes to the `make_http_request`

Reply:
Added test